src/main/java/RebornStorage/tiles/CraftingNode.java

```diff
@@ -0,0 +1,170 @@
+package RebornStorage.tiles;
+
+import RebornStorage.init.ModBlocks;
+import com.raoulvdberge.refinedstorage.api.autocrafting.ICraftingPattern;
+import com.raoulvdberge.refinedstorage.api.autocrafting.ICraftingPatternContainer;
+import com.raoulvdberge.refinedstorage.api.autocrafting.ICraftingPatternProvider;
+import com.raoulvdberge.refinedstorage.api.network.INetwork;
+import com.raoulvdberge.refinedstorage.api.network.node.INetworkNode;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraftforge.items.IItemHandler;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class CraftingNode implements INetworkNode, ICraftingPatternContainer {
+
+	TileMultiCrafter multiCrafter;
+	List<ICraftingPattern> patterns = new ArrayList<>();
+	@Nullable
+	INetwork network;
+	int ticks = 0;
+
+	public CraftingNode(TileMultiCrafter multiCrafter) {
+		this.multiCrafter = multiCrafter;
+	}
+
+	public void rebuildPatterns() {
+		this.patterns.clear();
+
+		ICraftingPatternContainer patternContainer = this;
+
+		if(multiCrafter.inv != null && multiCrafter.getMultiBlock() != null){
+			patterns.addAll(Arrays.stream(multiCrafter.inv.contents)
+				.filter(stack -> !stack.isEmpty())
+				.map(stack -> ((ICraftingPatternProvider)stack.getItem()).create(multiCrafter.getWorld(), stack, patternContainer))
```

Comment:
Do stack.copy() here, or you'll get problems if the stack is removed from the container and is transformed to an empty stack during the autocrafting.


So: ``.map(stack -> ((ICraftingPatternProvider)stack.getItem()).create(multiCrafter.getWorld(), stack.copy(), patternContainer))``